packages/xod-arduino/src/transpiler.js

```diff
@@ -556,10 +569,17 @@ const transformProjectWithImpls = def(
 
         return R.merge(
           {
-            config: {
-              XOD_DEBUG: liveness === LIVENESS.DEBUG,
-              XOD_SIMULATION: liveness === LIVENESS.SIMULATION,
-            },
+            config: R.merge(
+              {
+                XOD_DEBUG: liveness === LIVENESS.DEBUG,
+                XOD_SIMULATION: liveness === LIVENESS.SIMULATION,
+                XOD_USERNAME_NEEDED: doesContainSpecialStringLiterals(
```

Comment:
If a project `doesContainSpecialStringLiterals` it doesn’t mean that the `XOD_USERNAME_NEEDED`. Maybe, it’s just `XOD_PROJECT`.